### PR TITLE
Eskip lexer refactor

### PR DESCRIPTION
**Pull request by aryszka**

### CODE REVIEW

File: eskip/lexer.go [hunk 1]
```diff
@@ -17,218 +17,306 @@ package eskip
...
-// implements the lexer instance
+type charPredicate func(byte) bool
+
+type scanner interface {
+	scan(string) (token, string, error)
+}
+
+type scannerFunc func(string) (token, string, error)
+
+func (sf scannerFunc) scan(code string) (token, string, error) { return sf(code) }
```
Comment (lmineiro) [inline eskip/lexer.go]: I see what you did there ;-)

File: eskip/lexer.go [hunk 2]
```diff
@@ -17,218 +17,306 @@ package eskip
...
 
-// unescaping only '/'
-func convertRegexp(s string) string {
-	return unescape(s[1:len(s)-1], "/")
+func scanWhitespace(code string) string { return scanVoid(code, isWhitespace) }
+func scanComment(code string) string {
+	return scanVoid(code, func(c byte) bool { return !isNewline(c) })
 }
+func scanStringLiteral1(code string) (token, string, error) { return scanStringLiteral('"', code) }
+func scanStringLiteral2(code string) (token, string, error) { return scanStringLiteral('`', code) }
```
Comment (lmineiro) [inline eskip/lexer.go]: Uninspired naming? What was the rationale for the 1 and 2?

Comment (aryszka) [inline eskip/lexer.go]: the names scanStringLiteralDoubleQuote and scanStringLiteralBacktick felt just so.. something. Do you have a good late-wednesday suggestion?

Comment (lmineiro) [inline eskip/lexer.go]: `scanDoubleQuote` and `scanBacktick` :)

Comment (aryszka) [inline eskip/lexer.go]: thx, fixed ;)

**Pull request merged**